### PR TITLE
[CWS] Add new exit secl rule

### PR DESCRIPTION
**Pull request by YoannGh**
### What does this PR do?

The PR adds support for writing SECL rules for a process exit event.
 
It exposes:
- `exit.cause` : the cause of the process termination (EXITED, SIGNALED or COREDUMPED)
- `exit.code` : the exit code of the process, which is the signal number if the process terminated because of a signal or coredump; or the return value of the process if it terminated normally

For example: `exit.cause == SIGNALED && exit.code == SIGKILL && exit.file.path == "/usr/bin/ssh"`

Functional tests can be run with the following:
`inv -e security-agent.functional-tests --testflags "-test.v -test.run TestProcessExit"`

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/security/secl/model/model.go [hunk 1]
```diff
@@ -332,6 +333,14 @@ type ExecEvent struct {
 	*Process
 }
 
+// ExitEvent represents a process exit event
+//msgp:ignore ExecEvent
+type ExitEvent struct {
+	*Process
+	Cause uint32 `field:"cause" msg:"exit_cause"` // Cause of the process termination (one of EXITED, SIGNALED, COREDUMPED)
```
Comment (safchain) [inline pkg/security/secl/model/model.go:340]: nit: if we ignore `msgp:ignore` we can omit the msg tag here

**Review by safchain — APPROVED** [review]

**Review by lebauce — APPROVED** [review]

File: docs/cloud-workload-security/agent_expressions.md [hunk 2]
```diff
@@ -424,6 +425,61 @@ A process was executed or forked
...
+| `exit.envp` | string | Environment variables of the process |  |
+| `exit.envs` | string | Environment variable names of the process |  |
+| `exit.envs_truncated` | bool | Indicator of environment variables truncation |  |
+| `exit.euid` | int | Effective UID of the process |  |
+| `exit.euser` | string | Effective user of the process |  |
+| `exit.file.change_time` | int | Change time of the file |  |
+| `exit.file.filesystem` | string | File's filesystem |  |
+| `exit.file.gid` | int | GID of the file's owner |  |
+| `exit.file.group` | string | Group of the file's owner |  |
+| `exit.file.in_upper_layer` | bool | Indicator of the file layer, in an OverlayFS for example |  |
```
Comment (maycmlee) [inline docs/cloud-workload-security/agent_expressions.md:459]:
```suggestion
| `exit.file.in_upper_layer` | bool | Indicator of the file layer, for example, in an OverlayFS |  |
```

File: docs/cloud-workload-security/agent_expressions.md [hunk 3]
```diff
@@ -424,6 +425,61 @@ A process was executed or forked
 | `exec.uid` | int | UID of the process |  |
 | `exec.user` | string | User of the process |  |
 
+### Event `exit`
+
+A process terminated
```
Comment (maycmlee) [inline docs/cloud-workload-security/agent_expressions.md:430]:
```suggestion
A process was terminated
```

File: docs/cloud-workload-security/secl.json [hunk 4]
```diff
@@ -1308,6 +1308,303 @@
...
+        {
+          "name": "exit.file.group",
+          "type": "string",
+          "definition": "Group of the file's owner",
+          "constants": ""
+        },
+        {
+          "name": "exit.file.in_upper_layer",
+          "type": "bool",
+          "definition": "Indicator of the file layer, in an OverlayFS for example",
```
Comment (maycmlee) [inline docs/cloud-workload-security/secl.json:1471]:
```suggestion
          "definition": "Indicator of the file layer, for example, in an OverlayFS",
```

**Review by maycmlee — COMMENTED** [review]
A couple of suggestions!

**Review by maycmlee — APPROVED** [review]

**Pull request merged**